lib/ansible/modules/cloud/amazon/batch_job_definition.py

```diff
@@ -0,0 +1,493 @@
+        current_state = 'present'
+
+    if state == 'present':
+        if current_state == 'present':
+            # check if definition has changed and register a new version if necessary
+            if not job_definition_equal(module, current_job_definition):
+                create_job_definition(module, aws)
+                action_taken = 'updated with new version'
+                changed = True
+        else:
+            # Create Job definition
+            changed = create_job_definition(module, aws)
+            action_taken = 'added'
+
+        response = get_current_job_definition(module, aws)
+        if not response:
+            module.fail_json(msg='Unable to get job definition information after creating/updating')
+    else:
+        if current_state == 'present':
+            # remove the Job definition
+            changed = remove_job_definition(module, aws)
+            action_taken = 'deregistered'
+    return dict(changed=changed, ansible_facts=dict(batch_job_definition_action=action_taken), response=response)
+
+
+# ---------------------------------------------------------------------------------------------------
+#
+#   MAIN
+#
+# ---------------------------------------------------------------------------------------------------
+
+def main():
+    """
+    Main entry point.
+
+    :return dict: ansible facts
+    """
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            state=dict(required=False, default='present', choices=['present', 'absent']),
+            job_definition_name=dict(required=True, default=None),
+            type=dict(required=True, default=None),
+            parameters=dict(type='dict', default=None),
+            image=dict(required=True, default=None),
+            vcpus=dict(type='int', required=True, default=None),
+            memory=dict(type='int', required=True, default=None),
+            command=dict(type='list', default=[]),
+            job_role_arn=dict(default=None),
+            volumes=dict(type='list', default=[]),
+            environment=dict(type='list', default=[]),
+            mount_points=dict(type='list', default=[]),
+            readonly_root_filesystem=dict(default=None),
+            privileged=dict(default=None),
+            ulimits=dict(type='list', default=[]),
+            user=dict(default=None),
+            attempts=dict(type='int', default=None),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_together=[]
+    )
+
+    # validate dependencies
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required for this module.')
+
+    aws = AWSConnection(module, ['batch'])
+
+    validate_params(module, aws)
+
+    results = manage_state(module, aws)
+
+    module.exit_json(**results)
+
+# ansible import module(s) kept at ~eof as recommended
+from ansible.module_utils.basic import *
+from ansible.module_utils.ec2 import *
```

Comment:
Hm... I'm not sure why it fails for you if these are moved up to the top. By 'top' do you mean around [line 186](https://github.com/ansible/ansible/pull/24402/files#diff-576d1ebf1342b87bb8a863a1633e3311R186)? That is around where they should go. And
in addition, importing * is avoided now so things don't 'magically' work. Also you'll want to import HAS_BOTO3 rather than set it on  [line 182](https://github.com/ansible/ansible/pull/24402/files#diff-576d1ebf1342b87bb8a863a1633e3311R182) or line 184.
So you'll want something like `from ansible.module_utils.ec2 import ec2_argument_spec, get_aws_connection_info, boto3_conn, HAS_BOTO3`.